Libraries/Database/SessionExtensions.cs

```diff
@@ -24,33 +24,66 @@ public static class SessionExtensions
         /// <param name="queryExecutor">
         /// Delegate to execute a query in a custom way, based on the prepared inputs.
         /// </param>
+        /// <param name="transaction">If not <see langword="null"/> it must be an open DB transaction.</param>
         /// <returns>The result set of the query, rows mapped to <typeparamref name="TRow"/>.</returns>
         public static async Task<IEnumerable<TRow>> RawQueryAsync<TRow>(
             this ISession session,
             string sql,
             IDictionary<string, object> parameters = null,
-            Func<(string ParsedQuery, IDbTransaction Transaction), Task<IEnumerable<TRow>>> queryExecutor = null)
+            Func<(string ParsedQuery, IDbTransaction Transaction), Task<IEnumerable<TRow>>> queryExecutor = null,
+            DbTransaction transaction = null)
         {
-            var transaction = await session.DemandAsync();
+            transaction ??= await session.DemandAsync();
+            var query = GetQuery(sql, transaction, session, parameters);
 
+            return queryExecutor == null
+                ? await transaction.Connection.QueryAsync<TRow>(query, transaction: transaction)
+                : await queryExecutor((query, transaction));
+        }
+
+        /// <summary>
+        /// Executes a raw SQL string command that doesn't return data in a database-agnostic way by running it
+        /// through Orchard's <see cref="SqlParser"/>.
+        /// </summary>
+        /// <param name="sql">
+        /// The raw SQL string. Doesn't need to use table prefixes or care about SQL dialects.
+        /// </param>
+        /// <param name="parameters">Input parameters passed to the query.</param>
+        /// <param name="transaction">If not <see langword="null"/> it must be an open DB transaction.</param>
+        /// <returns>The number of rows affected.</returns>
+        public static async Task<int> RawExecuteNonQuery(
+            this ISession session,
+            string sql,
+            IDictionary<string, object> parameters = null,
+            DbTransaction transaction = null)
+        {
+            transaction ??= await session.DemandAsync();
+            var query = GetQuery(sql, transaction, session, parameters);
+
+            return await transaction.Connection.ExecuteAsync(query, transaction: transaction);
+        }
+
+        private static string GetQuery(
+            string sql,
+            DbTransaction transaction,
+            ISession session,
+            IDictionary<string, object> parameters)
+        {
             var parserResult = SqlParser.TryParse(
                 sql,
                 TransactionSqlDialectFactory.For(transaction),
                 session.Store.Configuration.TablePrefix,
                 parameters,
                 out var query,
                 out var messages);
+            if (parserResult) return query;

```

Comment:
Elé kellene egy sortörés, hogy kicsit szellősebb legyen, ahogy már beleraktam korábban :).

Reply:
Miért? A két utasítás logikailag összetartozó (ugye mehetne az egész SqlParser.TryParse rögtön az if-be csak az csúnya lenne). Meg akkor már minden utasítás külön szigetet alkotna amitől szerintem az egész értelmét veszti.

Reply:
Igen, de itt van egy hétsoros metódushívás, amivel nagyon egybefolyik.